### PR TITLE
:zap: Table_blocking feature

### PR DESCRIPTION
**Pull request by KolushovAlexandr**

### CODE REVIEW

File: pos_multi_session_restaurant/static/src/js/pos_multi_session_restaurant.js [hunk 1]
```diff
@@ -372,4 +394,32 @@ odoo.define('pos_multi_session_restaurant', function(require){
...
+                this._super();
+            } else {
+                var table = this.table;
+                if (table) {
+                    if (this.pos.check_table_admittance(table)) {
+                        this._super();
+                    } else {
+                        this.pos.gui.show_popup('confirm',{
+                            'title': _t('Can not open the Table'),
+                            'body': _t('The Table is served by another waiter'),
```
Comment (yelizariev) [inline pos_multi_session_restaurant/static/src/js/pos_multi_session_restaurant.js]: Specify in the message by whom is served

File: pos_multi_session_restaurant/static/src/js/pos_multi_session_restaurant.js [hunk 2]
```diff
@@ -220,7 +220,29 @@ odoo.define('pos_multi_session_restaurant', function(require){
...
+                return true;
+            }
+            var table_orders = this.get_table_orders(table);
+            var cashier = this.cashier;
+            var user_is_manager = _.contains(this.get_cashier().groups_id, this.config.group_pos_manager_id[0]);
+            var creator_ids = _.chain(table_orders)
+                .map(function(o){
+                    return o.ms_info.created.user.id;
+                })
+                .unique().value();
```
Comment (yelizariev) [inline pos_multi_session_restaurant/static/src/js/pos_multi_session_restaurant.js:240]: You don't need to compute it when ``user_is_manager`` is true

File: pos_multi_session_restaurant/static/src/js/pos_multi_session_restaurant.js [hunk 3]
```diff
@@ -372,4 +394,32 @@ odoo.define('pos_multi_session_restaurant', function(require){
...
+    floors.TableWidget.include({
+        click_handler: function(){
+            var self = this;
+            var floorplan = this.getParent();
+            if (floorplan.editing) {
+                this._super();
+            } else {
+                var table = this.table;
+                if (table) {
+                    if (this.pos.check_table_admittance(table)) {
```
Comment (yelizariev) [inline pos_multi_session_restaurant/static/src/js/pos_multi_session_restaurant.js]: You can simplify nesting
```
if (!table or this.pos.check_table_admittance(table)){
return  super;
}
```

**Review by yelizariev — CHANGES_REQUESTED** [review]

**Review by gustavovalverde — APPROVED** [review]
Just add the translation file.

File: pos_multi_session_restaurant/static/src/js/pos_multi_session_restaurant.js [hunk 4]
```diff
@@ -372,4 +397,31 @@ odoo.define('pos_multi_session_restaurant', function(require){
...
+                    return this._super();
+                }
+                var table_waiters = this.pos.check_table_inaccessibility(table);
+                if (!table_waiters || !table_waiters.length) {
+                    return this._super();
+                }
+                return this.pos.gui.show_popup('confirm',{
+                    'title': _t('Can not open the Table'),
+                    'body': _t('The Table is served by another waiter ' + _.pluck(table_waiters, 'name')),
+                    confirm: function(){
```
Comment (Ramil-Mukhametzyanov) [inline pos_multi_session_restaurant/static/src/js/pos_multi_session_restaurant.js]: It does not make sense: cancel or confirm
Why not just 'ok'?

**Review by Ramil-Mukhametzyanov — APPROVED** [review]

**Review by yelizariev — APPROVED** [review]

**Pull request merged**